src/bucket/BucketIndexImpl.h

```diff
@@ -45,7 +49,7 @@ template <class IndexT> class BucketIndexImpl : public BucketIndex
         void
         load(Archive& ar)
         {
-            ar(keysToOffset, filter);
+            ar(assetToPoolID, keysToOffset, filter);
```

Comment:
Is there a risk of loading the old version with the new code (or vice versa)?

Reply:
No, in `BucketIndex::load`, we load the version and pageSize first. These two values are always serialized first no matter the index version. Then we check the index version, and if the on-disk version number doesn't match the expected version number, we drop the on-disk index and reindex from scratch.

Reply:
should we delete the outdated index right away instead of keeping both until the bucket gets cleaned up? (so that we don't accidentally use it)

Reply:
We don't actually have both indexes persisted, but we generate a new index, then rename the new index the the same name as the old index. On linux, this deletes the old file. However, I looked up in the standard and this behavior is implementation defined, so I agree we should delete explicitly here.